### ISSUE TITLE
package_create raises KeyError with `return_id_only` in context

### ISSUE THREAD

**Issue by metaodi** [ckan/ckan#4418]
### CKAN Version if known (or site URL)
master

### Please describe the expected behaviour
Calling `package_create` with the `return_id_only` in the context should create a package and return the id.

### Please describe the actual behaviour

I noticed this in a [build of one of my extensions](https://travis-ci.org/opendatazurich/ckanext-stadtzh-harvest/jobs/418714960#L4319-L4330) against the `master` of ckan.

Calling `package_create` with the `return_id_only` in the context raises a `KeyError`

```
ckanext.harvest.model: DEBUG: Unable to get content for package: test_dataset: KeyError('id',) / Traceback (most recent call last):
  File "/home/travis/build/opendatazurich/ckanext-stadtzh-harvest/ckanext/stadtzhharvest/harvester.py", line 268, in import_stage
    self._import_package(harvest_object)
  File "/home/travis/build/opendatazurich/ckanext-stadtzh-harvest/ckanext/stadtzhharvest/harvester.py", line 315, in _import_package
    dataset_id = self._create_package(package_dict, harvest_object)
  File "/home/travis/build/opendatazurich/ckanext-stadtzh-harvest/ckanext/stadtzhharvest/harvester.py", line 560, in _create_package
    p.toolkit.get_action('package_create')(context, dataset)
  File "/home/travis/build/opendatazurich/ckanext-stadtzh-harvest/ckan/ckan/logic/__init__.py", line 464, in wrapped
    result = _action(context, data_dict, **kw)
  File "/home/travis/build/opendatazurich/ckanext-stadtzh-harvest/ckan/ckan/logic/action/create.py", line 230, in package_create
    return context['id']
KeyError: 'id'
```

The problem was introduced with the PR #4315:

https://github.com/ckan/ckan/blob/ba3a2201793c21ef62d8ee03e5ba36b1d754a159/ckan/logic/action/create.py#L229-L230

Here `context['id']` should be replaced by `pkg.id`.

### What steps can be taken to reproduce the issue? 

Call `package_create` with the `return_id_only` in the context set to `True`.

**Comment by metaodi** [comment]
I will create a PR for this.

**Comment by amercader** [comment]
Fixed by #4429
